source/Applications/Basic/Visualization/CaptureWritePCLVis3D/CaptureWritePCLVis3D.cpp

```diff
@@ -1,21 +1,120 @@
+        {
+            pointCloud->points[i].normal_x = normals(i).x; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            pointCloud->points[i].normal_y = normals(i).y; // NOLINT(cppcoreguidelines-pro-type-union-access)
+            pointCloud->points[i].normal_z = normals(i).z; // NOLINT(cppcoreguidelines-pro-type-union-access)
+        }
+
+        return pointCloud;
+    }
+
+    void addPointCloudToViewer(boost::shared_ptr<pcl::visualization::PCLVisualizer> &viewer,
+                               const boost::shared_ptr<pcl::PointCloud<pcl::PointXYZRGBNormal>> &pointCloud)
+    {
+        viewer->addPointCloud<pcl::PointXYZRGBNormal>(pointCloud);
+
+        const int normalsSkipped = 10;
+        std::cout << "Note! 1 out of " << normalsSkipped << " normals are visualized" << std::endl;
+        viewer->addPointCloudNormals<pcl::PointXYZRGBNormal, pcl::PointXYZRGBNormal>(pointCloud,
+                                                                                     pointCloud,
+                                                                                     normalsSkipped,
+                                                                                     1,
+                                                                                     "normals");
+    }
+
+    void addPointCloudToViewer(boost::shared_ptr<pcl::visualization::PCLVisualizer> &viewer,
+                               const boost::shared_ptr<pcl::PointCloud<pcl::PointXYZRGB>> &pointCloud)
+    {
+        viewer->addPointCloud<pcl::PointXYZRGB>(pointCloud);
+    }
+
+    template<typename T>
+    void visualizePointCloudPCL(const boost::shared_ptr<pcl::PointCloud<T>> &pointCloud)
+    {
+        auto viewer = boost::make_shared<pcl::visualization::PCLVisualizer>("Viewer");
+
+        addPointCloudToViewer(viewer, pointCloud);
+
+        viewer->setCameraPosition(0, 0, -100, 0, -1, 0);
+
+        std::cout << "Press r to centre and zoom the viewer so that the entire cloud is visible" << std::endl;
+        std::cout << "Press q to me exit the viewer application" << std::endl;
+        while(!viewer->wasStopped())
+        {
+            viewer->spinOnce(100);
+            std::this_thread::sleep_for(std::chrono::milliseconds(100));
+        }
+    }
+
+    template<typename T>
+    void visualizeAndSavePointCloudPCL(const boost::shared_ptr<pcl::PointCloud<T>> &pointCloud)
+    {
+        std::cout << "Visualizing PCL point cloud" << std::endl;
+        visualizePointCloudPCL(pointCloud);
+
+        std::string pointCloudFile = "Zivid3D.pcd";
+        std::cout << "Saving point cloud to file: " << pointCloudFile << std::endl;
+        pcl::io::savePCDFileBinary(pointCloudFile, *pointCloud);
+        std::cerr << "Saved " << pointCloud->points.size() << " points" << std::endl;
+    }
+
+} // namespace
+
+int main(int argc, char **argv)
 {
     try
     {
+        bool useNormals = (argc >= 2 && std::string(argv[1]) == "normals") ? true : false;
```

Comment:
```suggestion
        bool useNormals = argc >= 2 && std::string(argv[1]) == "normals";
```

The ternary operator is redundant here